src/consumeroffset/kafka_offset_collect_test.go

```diff
@@ -220,13 +215,93 @@ func TestCollectOffsetsForConsumerGroup(t *testing.T) { // nolint: funlen
 					assert.Equal(t, testClientID, entity.Metrics[0].Metrics["clientID"])
 					assert.Equal(t, topicOne, entity.Metrics[0].Metrics["topic"])
+				default:
+					assert.Fail(t, "not expected")
 				}
 				assert.NotEmpty(t, entity)
 			}
 		})
 	}
 }
 
+func TestNoActiveConsumersForConsumerGroup(t *testing.T) { // nolint: funlen
+	// MemberAssignment mock created as in sarama's consumer_group_member_test.go
+	members := map[string]*sarama.GroupMemberDescription{
+		testClientID: {
+			ClientId:       testClientID,
+			ClientHost:     "a-host",
+			MemberMetadata: nil,
+			MemberAssignment: []byte{
+				0, 0, // Version
+				0, 0, 0, 0, // Topic array length
+				0, 0, 0, 3, 0x01, 0x02, 0x03, // Userdata
+			},
+		},
+	}
+
+	args.GlobalArgs = &args.ParsedArguments{}
+
+	expectedTopicEntities := map[string]map[string]float64{
+		topicOne: {
+			// comes from: 25 - 10 + 30 - 10
+			"totalLag": 35,
+			// comes from max: 30 - 10
+			"maxLag":          20,
+			"activeConsumers": 0,
+		},
+		topicTwo: {
+			// comes from: 25 - 10 + 30 - 10
+			"totalLag": 35,
+			// comes from max: 30 - 10
+			"maxLag":          20,
+			"activeConsumers": 0,
+		},
+	}
+
+	expectedCGroupEntities := map[string]map[string]float64{
+		consumerGroupOne: {
+			// comes from: (25 - 10 + 30 - 10) + (25 - 10 + 30 - 10)
+			"totalLag": 70,
+			// comes from max: 30 - 10
+			"maxLag":          20,
+			"activeConsumers": 0,
+		},
+	}
+	expectedNumEntities := 3
+
+	args.GlobalArgs.InactiveConsumerGroupOffset = true
+	args.GlobalArgs.ConsumerGroupOffsetByTopic = true
+
+	kafkaIntegration, _ := integration.New("test", "test")
+
+	collectOffsetsForConsumerGroup(
+		&ConsumerGroupTopicListerMock{},
+		consumerGroupOne,
+		members,
+		kafkaIntegration,
+		&TopicOffsetGetterMock{},
+	)
+
+	assert.Equal(t, expectedNumEntities, len(kafkaIntegration.Entities))
+	for _, entity := range kafkaIntegration.Entities {
+		switch entity.Metadata.Namespace {
+		case nrConsumerGroupEntity:
+			if entity.Metrics[0].Metrics["consumerGroup"] == consumerGroupOne {
+				assert.Equal(t, expectedCGroupEntities[consumerGroupOne]["totalLag"], entity.Metrics[0].Metrics["consumerGroup.totalLag"])
+				assert.Equal(t, expectedCGroupEntities[consumerGroupOne]["maxLag"], entity.Metrics[0].Metrics["consumerGroup.maxLag"])
+				assert.Equal(t, expectedCGroupEntities[consumerGroupOne]["activeConsumers"], entity.Metrics[0].Metrics["consumerGroup.activeConsumers"])
+			}
+		case nrConsumerGroupTopicEntity:
+			topicName := fmt.Sprintf("%v", entity.Metrics[0].Metrics["topic"])
+			assert.Equal(t, expectedTopicEntities[topicName]["totalLag"], entity.Metrics[0].Metrics["consumerGroup.totalLag"])
+			assert.Equal(t, expectedTopicEntities[topicName]["maxLag"], entity.Metrics[0].Metrics["consumerGroup.maxLag"])
+			assert.Equal(t, expectedTopicEntities[topicName]["activeConsumers"], entity.Metrics[0].Metrics["consumerGroup.activeConsumers"])
+		default:
+			assert.Fail(t, "not expected")
+		}
+	}
+}
```

Comment:
I'm not sure if a brief explanation of what entities are not generated when there are no active consumers could help here.